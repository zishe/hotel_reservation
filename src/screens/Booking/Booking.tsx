import React from "react";
import { Text } from "react-native";
import { Container, Content, Body, ListItem, List, Left } from "native-base";
import { AppHeader } from "../../components";
import styles from "./styles";

interface Props {
  navigation: any;
}

export default class BookingScreen extends React.PureComponent<Props, {}> {
  static navigationOptions = {
    drawerLabel: (): any => null
  };

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.screenContainer}>
          <AppHeader
            title="Reservation Details"
            backScreen="Bookings"
            navigation={this.props.navigation}
          />
          <List>{this.renderDetails()}</List>
        </Content>
      </Container>
    );
  }

  renderDetails = () => {
    const reservation: any = this.props.navigation.getParam("reservation", {});

    const fields = [
      { label: "Reservation Id", prop: "id" },
      { label: "Guest Name", prop: "name" },
      { label: "Hotel", prop: "hotelName" },
      { label: "Arrival", prop: "arrivalDate" },
      { label: "Departure", prop: "departureDate" }
    ];

    return fields.map(field => {
      return this.renderDetailsItem(field.label, reservation[field.prop]);
    });
  };

  renderDetailsItem = (label: string, val: string) => {
    const itemKey = `reservationdetails-${label}-${val}`;
    return (
      <ListItem key={itemKey}>
        <Left>
          <Text>{label}:</Text>
        </Left>
        <Body>
          <Text>{val}</Text>
        </Body>
      </ListItem>
    );
  };
}
