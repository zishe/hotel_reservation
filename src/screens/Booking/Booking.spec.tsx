import React from "react";
import renderer from "react-test-renderer";
import Booking from "./Booking";

const reservation = {
  id: "1",
  name: "TestName",
  hotelName: "TestHotel",
  arrivalDate: "03/09/2019",
  departureDate: "03/16/2019"
};

const mockProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
    getParam: jest.fn().mockReturnValue(reservation)
  },
  ...props
});

describe("BookingsScreen", () => {
  /**
   * @description test component renders
   */
  it("should render as expected", () => {
    const props = mockProps({});
    const booking = renderer.create(<Booking {...props} />).toJSON;
    expect(booking).toMatchSnapshot();
  });
});
