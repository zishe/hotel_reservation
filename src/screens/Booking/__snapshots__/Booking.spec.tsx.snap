// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingsScreen should render as expected 1`] = `[Function]`;
